### PR TITLE
gh-110828: AIX 32bit build needs -latomic for _testcapi module

### PR DESCRIPTION
**Pull request by ayappanec**
This PR is to fix the problem mentioned in the issue #110828 

* Issue: gh-110828

### CODE REVIEW

**Comment by erlend-aasland** [comment]
cc. @colesbury

File: Misc/NEWS.d/next/Build/2023-10-17-03-10-40.gh-issue-110828.31vQ9B.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+AIX 32bit build needs -latomic for _testcapi module build
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Build/2023-10-17-03-10-40.gh-issue-110828.31vQ9B.rst]:
```suggestion
AIX 32bit needs ``-latomic`` to build the :mod:`!_testcapi` extension module.
```

File: configure.ac [hunk 2]
```diff
@@ -7323,7 +7324,10 @@ PY_STDLIB_MOD([_hashlib], [], [test "$ac_cv_working_openssl_hashlib" = yes],
   [$OPENSSL_INCLUDES], [$OPENSSL_LDFLAGS $OPENSSL_LDFLAGS_RPATH $LIBCRYPTO_LIBS])
 
 dnl test modules
-PY_STDLIB_MOD([_testcapi], [test "$TEST_MODULES" = yes])
+PY_STDLIB_MOD([_testcapi],
+    [test "$TEST_MODULES" = yes],  []
```
Comment (erlend-aasland) [inline configure.ac]: Either you missed a comma, or this quote is superfluous :)

```suggestion
    [test "$TEST_MODULES" = yes],
```

Comment (ayappanec) [reply configure.ac]: I just maintained the same how it was before #109344

**Review by colesbury — COMMENTED** [review]
This LGTM fwiw.

**Review by colesbury — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the report and fix, @ayappanec; thanks for the review, @colesbury!

**Comment by ayappanec** [comment]
Thanks @erlend-aasland for your time in reviewing, fixing & merging the fix. Thanks too @colesbury for reviewing & approving the changes.

**Pull request merged**